connectproxy.go

```diff
@@ -246,7 +247,7 @@ func (cd *connectDialer) Dial(network, addr string) (net.Conn, error) {
 	}
 	req.Close = false
 	if cd.haveAuth {
-		req.SetBasicAuth(cd.username, cd.password)
+		req.Header.Set("Proxy-Authorization", basicAuth(cd.username, cd.password))
 	}
 	req.Header = cd.config.Header
```

Comment:
I think you want mo move `req.Header = cd.config.Header` a bit up, like:
```
if cd.config.Header != nil {
	req.Header = cd.config.Header	
}

if cd.haveAuth {
    req.Header.Set("Proxy-Authorization", basicAuth(cd.username, cd.password))
}
```

Because else the header will always be overwritten by the value specified in the config, which is nil in some scenarios.